pkg/deployment/resources/pdbs.go

```diff
@@ -0,0 +1,105 @@
+package resources
+
+import (
+	"fmt"
+
+	api "github.com/arangodb/kube-arangodb/pkg/apis/deployment/v1alpha"
+	"github.com/arangodb/kube-arangodb/pkg/util/k8sutil"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func min(a int, b int) int {
+	if a > b {
+		return a
+	}
+	return b
+}
+
+// EnsurePDBs ensures Pod Distruption Bugets for different server groups in Cluster mode
+func (r *Resources) EnsurePDBs() error {
+
+	// Only in Cluster and Production Mode
+	spec := r.context.GetSpec()
+	if spec.IsProduction() && spec.GetMode().IsCluster() {
+
+		minAgents := spec.GetServerGroupSpec(api.ServerGroupAgents).GetCount() - 1
```

Comment:
Please add comment to explain these rules.
This is probably the key part of the entire PR